### PR TITLE
[Bug] Adding Adjustment to Route C of manage_versions.py

### PR DESCRIPTION
**Pull request by terrancedejesus**
## Issues
* https://github.com/elastic/detection-rules/pull/2272

## Summary
Adjusted route C to update the dictionary for the `lock_from_file` variable as `max_allowable_version` was overwritten and therefore did not exist in the output.

Testing:
1. Check out 8.2
2. Update `command_and_control_telnet_port_activity.toml` in rule contents
3. Run `python -m detection_rules dev build-release --update-version-lock`
4. Check diff for `version.lock.json` file and notice `max_allowable_version` is missing

Bug Screenshot:
Screen Shot 2022-09-19 at 2 43 17 PM

Solution Screenshot:
Screen Shot 2022-09-19 at 2 19 27 PM

### CODE REVIEW

**Review by Mikaayenson — APPROVED** [review]
LGTM

**Review by brokensound77 — APPROVED** [review]
LGTM - after merging can you run a test lock

**Pull request merged**
